### ISSUE TITLE
Docker for Windows won't start, key already exists

### ISSUE THREAD

**Issue by Scroovy** [docker/for-win#1938]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have submitted Diagnostics
  - [x] I have included the Diagnostics ID in this Issue
**6DA69920-40C1-4EED-AF10-CDF6A48C7286/2018-04-11_17-07-39**
  - Windows Version: 10 Pro
  - Docker for Windows Version: Version 18.03.0-ce-win59 (16762)

### Expected behavior
Starting docker successfully
### Actual behavior
Docker crashes with following error
### Information

[log.txt](https://github.com/docker/for-win/files/1899418/log.txt)

**Error message:**
Ein Element mit dem gleichen Schlüssel wurde bereits hinzugefügt.
_(Translation: An item with the same key has already been added.)_
   bei System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   bei System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   bei Docker.Backend.Processes.ApiProxy.GetEnvVars() in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\Processes\ApiProxy.cs:Zeile 55.
   bei Docker.Backend.Processes.ApiProxy.StartWinNamedPipe(Settings settings, String daemonNamedPipe) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\Processes\ApiProxy.cs:Zeile 72.
   bei Docker.Backend.ContainerEngine.Windows.DoStart(Settings settings, String daemonOptions) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\ContainerEngine\Windows.cs:Zeile 218.
   bei Docker.Backend.ContainerEngine.Windows.Start(Settings settings, String daemonOptions) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\ContainerEngine\Windows.cs:Zeile 93.
   bei Docker.Core.Pipe.NamedPipeServer.<>c__DisplayClass9_0.<Register>b__0(Object[] parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeServer.cs:Zeile 46.
   bei Docker.Core.Pipe.NamedPipeServer.RunAction(String action, Object[] parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeServer.cs:Zeile 144.

I already tried:
Uninstalling docker & reinstalling (newest stable and edge version)
Disabling & Enabling Hyper-V
Deleting the Program Data Docker Files
Restarting everything several times

Please I need help!

**Comment by firefrorefiddle** [comment]
I have the same problem.

**Comment by Scroovy** [comment]
Found the relating code:

  ```
 private IDictionary<string, string> GetEnvVars()
    {
      Dictionary<string, string> dictionary = new Dictionary<string, string>();
      foreach (DictionaryEntry environmentVariable in Environment.GetEnvironmentVariables())
      {
        string key;
        string str;
        if ((key = environmentVariable.Key as string) != null && (str = environmentVariable.Value as string) != null)
          dictionary.Add(key, str);
      }
      using (this._recorder.Impersonate())
      {
        string path1 = Paths.FullPath(Paths.UserProfile);
        dictionary.Add("HOME", path1);
        if (!dictionary.ContainsKey("KUBECONFIG"))
          dictionary.Add("KUBECONFIG", Path.Combine(path1, ".kube", "config"));
      }
      return (IDictionary<string, string>) dictionary;
    }
```

**Comment by stephen-turner** [comment]
Closing as dupe of #1701.

**Comment by stephen-turner** [comment]
Already fixed in 18.03.1.
